rocAL/include/readers/image/caffe_lmdb_record_reader.h

```diff
@@ -68,6 +68,8 @@ class CaffeLMDBRecordReader : public Reader {
 
     CaffeLMDBRecordReader();
 
+    size_t last_batch_padded_size() override; // The size of the number of samples padded in the last batch
```

Comment:
I think we should consider merging the caffe and caffe2 classes. Or derive caffe2 from caffe. I see lots of repeated member variables in both classes